torchtext/datasets/text_classification.py

```diff
@@ -373,6 +408,122 @@ def AmazonReviewFull(*args, **kwargs):
     return _setup_datasets(*(("AmazonReviewFull",) + args), **kwargs)
 
 
+def _generate_imdb_data(key, extracted_files, vocab, tokenizer, removed_tokens):
```

Comment:
This function is very complex for what it does.

Reply:
Merge it with `_setup_datasets`.